compose.override.dist.yaml

```diff
@@ -4,14 +4,14 @@ services:
   php-7.4: &php
     build:
       target: php
-      args: &base_build_args
```

Comment:
FYI: this was added because when you need to pass additional build args in `compose.override.yaml`, and you don't want to copy-paste same args in multiple services, you need to do something like this:

```yaml
  php-8.3:
    <<: *php
    build:
      args:
        <<: *base_build_args
        PHP_VERSION: "8.3.0"
        PHP_XDEBUG_VERSION: "3.3.0"
```

It's because args are merged on `compose.yaml` + `compose.override.yaml` level when combining files, but overriding is done _after_ `<<: *php` is resolved, so additional args defined in `php-7.4` are not merged with additional args defined in `php-8.3`. You need to do the "merge" manually by referencing anchor, on `compose.override.yaml` file level.

I know it's not used within the `compose.override.dist.yaml`, but instead of removing it, I would rather include the example usage everywhere so issue with overridden build args won't pop up again.

And it's not related to `host.docker.internal` 😉.

Reply:
I'm not sure I understand the reasons why this was added but indeed this is not related to the main change I wanted to submit in this PR so I reverted the removal.